ironfish-rust/src/public_account/description.rs

```diff
@@ -0,0 +1,218 @@
+pub struct PublicAccountDescription {
+    /// TODO(jwp): do we need a separate account address for later migration
+    /// pub(crate) account_address: EthereumAddress
+    pub(crate) version: u8,
+    // Minimum number of required signers
+    pub(crate) threshold: i16,
+    // Signers of the public account
+    pub(crate) signers: Vec<VerifyingKey>,
+    // Signatures of the signers for a given message
+    pub(crate) signatures: Vec<Signature>,
+    // asset transfers
+    pub(crate) transfers: Vec<Transfer>,
+    // address of the account
+    pub(crate) address: PublicAddress,
+}
+
+impl PublicAccountDescription {
+    pub fn read<R: io::Read>(mut reader: R) -> Result<Self, IronfishError> {
+        let mut version_buf = [0; 1];
+        reader.read_exact(&mut version_buf)?;
+        let version = version_buf[0];
+        
+        let mut threshold_buf = [0; 2];
+        reader.read_exact(&mut threshold_buf)?;
+        let threshold = i16::from_le_bytes(threshold_buf);
+
+        let mut signers_len_buf = [0; 2];
+        reader.read_exact(&mut signers_len_buf)?;
+        let signers_len = i16::from_le_bytes(signers_len_buf) as usize;
+
+        let mut signers = Vec::with_capacity(signers_len);
+        for _ in 0..signers_len {
+            let mut signer_buf = [0; 32];
+            reader.read_exact(&mut signer_buf)?;
+            let signer = VerifyingKey::from_bytes(&signer_buf)
+                .map_err(|_| IronfishError::new(IronfishErrorKind::InvalidData))?;
+            signers.push(signer);
+        }
+
+        let mut signatures_len_buf = [0; 2];
+        reader.read_exact(&mut signatures_len_buf)?;
+        let signatures_len = i16::from_le_bytes(signatures_len_buf) as usize;
+
+        let mut signatures = Vec::with_capacity(signatures_len);
+        for _ in 0..signatures_len {
+            let mut signature_buf = [0; 64];
+            reader.read_exact(&mut signature_buf)?;
+            let signature = Signature::from_bytes(&signature_buf);
+            signatures.push(signature);
+        }
+
+        let mut transfers_len_buf = [0; 2];
+        reader.read_exact(&mut transfers_len_buf)?;
+        let transfers_len = i16::from_le_bytes(transfers_len_buf) as usize;
+
+        let mut transfers = Vec::with_capacity(transfers_len);
+        for _ in 0..transfers_len {
+            let transfer = Transfer::read(&mut reader)?;
+            transfers.push(transfer);
+        }
+        let address = PublicAddress::read(&mut reader)?;
+
+
+        Ok(Self {
+            version,
+            threshold,
+            signers,
+            signatures,
+            address,
+            transfers,
+        })
+    }
+
+    pub fn write<W: io::Write>(&self, mut writer: W) -> Result<(), IronfishError> {
+        // TODO: think about ideal ordering here
+        writer.write_all(&self.version.to_le_bytes())?;
+        
+        writer.write_all(&self.threshold.to_le_bytes())?;
+
+        let signers_len = self.signers.len() as i16;
```

Comment:
Note that the result of `.len()` may not fit into a `u16`. It'd be best to use `.len().try_into()` and return an error in the unlikely event that the vector is too long

Reply:
Good call, I would have expected rust to catch this, seems like it is better to avoid explicit type casting for this reason.